### PR TITLE
Use GitHub Actions for python-client (instead of Jenkins)

### PR DESCRIPTION
**Pull request by terminalstderr**

### CODE REVIEW

**Comment by terminalstderr** [comment]
I really dig these github action workflows! The declerative style and YAML syntax remind me very much of Azure DevOps. (I do genuinely appreciate Microsoft's engineering in this DevOps space 😜)

**Review by morrowc — APPROVED** [review]
looks great... one question about pypi update at 'any update' or 'just python directory' ... which isn't blocking but for my knowledge :)

File: .github/workflows/python-client.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: Test Annotate Plugin
+        run: pip install pytest-github-actions-annotate-failures
+
+      - name: Test
+        run: |
+          pytest $WORKDIR
+        
+      - name: Publish distribution 📦 to PyPI
+        if: github.ref == 'refs/heads/main'
```
Comment (morrowc) [inline .github/workflows/python-client.yml:47]: this checks 'if we're in main, for a merge/etc', but doesn't check the directory being changed.
Maybe this all only runs if the python parts are changing though?

Comment (morrowc) [reply .github/workflows/python-client.yml:47]: Oh yea, probably lines 9 && 14 answer my question :)

**Pull request merged**